vnext/Microsoft.ReactNative.Cxx/ReactDispatcher.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#ifndef MICROSOFT_REACTNATIVE_REACTDISPATCHER
+#define MICROSOFT_REACTNATIVE_REACTDISPATCHER
+
+#include <winrt/Microsoft.ReactNative.h>
+#include "ReactHandleHelper.h"
+
+namespace winrt::Microsoft::ReactNative {
+
+// Represents a dispatcher queue to invoke work item asynchronously.
+// It wraps up the IReactDispatcher and adds convenience methods for
+// working with C++ types.
+struct ReactDispatcher {
+  ReactDispatcher(std::nullptr_t = nullptr) noexcept {}
+
+  explicit ReactDispatcher(IReactDispatcher const &handle) noexcept : m_handle{handle} {}
+
+  IReactDispatcher const &Handle() const noexcept {
+    return m_handle;
+  }
+
+  explicit operator bool() const noexcept {
+    return static_cast<bool>(m_handle);
```

Comment:
consider making this explicit by comparing against nullptr instead

Reply:
I will change it to code that C++/WinRT uses:
m_handle ? true : false;

Comparing with nullptr is more expensive as it involves calling a constructor with nullptr parameter and then comparing two objects.

---
In reply to: [427648694](https://github.com/microsoft/react-native-windows/pull/4953#discussion_r427648694) [](ancestors = 427648694)